tests/c_api_tests/test_cpp_api_model.cpp

```diff
@@ -83,34 +93,49 @@ TEST_CASE("C++ API Model") {
     int8_t load_type = 2;
     double load_p_specified = 0.0;
     double load_q_specified = 500.0;
-    Buffer load_buffer{PGM_def_input_sym_load, 1};
+    Buffer const load_buffer{PGM_def_input_sym_load, 1};
     load_buffer.set_value(PGM_def_input_sym_load_id, &load_id, -1);
     load_buffer.set_value(PGM_def_input_sym_load_node, &load_node, -1);
     load_buffer.set_value(PGM_def_input_sym_load_status, &load_status, -1);
     load_buffer.set_value(PGM_def_input_sym_load_type, &load_type, -1);
     load_buffer.set_value(PGM_def_input_sym_load_p_specified, &load_p_specified, -1);
     load_buffer.set_value(PGM_def_input_sym_load_q_specified, &load_q_specified, -1);
 
-    // add buffers
-    input_dataset.add_buffer("node", 1, 1, nullptr, node_buffer);
+    // add buffers - row
     input_dataset.add_buffer("sym_load", 1, 1, nullptr, load_buffer);
     input_dataset.add_buffer("source", 1, 1, nullptr, source_buffer);
 
+    // add buffers - columnar
+    input_dataset.add_buffer("node", 2, 2, nullptr, nullptr);
```

Comment:
After seeing this, it makes me wonder if `add_buffer` should have an overload specific for columnar like
```
void add_buffer(std::string const& component, Idx elements_per_scenario, Idx total_elements, Idx const* indptr) const {
        add_buffer(*this, component, elements_per_scenario, total_elements, indptr, nullptr);
    }
```
Later it can be removed as this API is not stable, but it may make the tests cleaner and easier to write for the user. Just food for thought.

Reply:
Having an overload is a valid approach, but here in this case I think the argument behind is that the user could simply provide a buffer for row based and a `nullptr` for columnar ones. The benefit of this one can be that the columnar input is explicit assumed and that debugging might be easier if not seeing corresponding `add_attr_buffer` calls later on. The `nullptr` here explicitly acts as a token for it.

Reply:
Just something to think about. But it can be brought up again when migrating the tests to the API, if needed. Feel free to resolve.

Reply:
I understand. Since we still have time before we publish this, what do you think? @TonyXiang8787

Reply:
The CPP wrapper is not stable anyway. I propose to leave it out of the scope of this PR, which is the C-API adjustment.

@figueroa1395 if you think this is helpful for the test, feel free to add this function in your PR.